mitodl/mit-open
# Add certificate as a real database field to LearningResource

### What are the relevant tickets?
- Closes https://github.com/mitodl/hq/issues/3970
- Will partially address https://github.com/mitodl/hq/issues/4082 (some n+1 queries were due to on-the-fly calculation of certification property).

### Description (What does it do?)
Makes `certificate` a real db field for `LearningResource` instead of a property, and adds calculation of the appropriate value for it to ETL pipelines (fwiw, the old calculation in the property was sometimes wrong).


### How can this be tested?
- Copy .env values for `PROLEARN_*`, `MITX_ONLINE_*`, and `EDX_API_*` from heroku.
- Run the migration, either via `docker compose up` or `./manage.py migrate`
- Run the following:
   ```
   ./manage.py backpopulate_prolearn_data
   ./manage.py backpopulate_mitxonline_data
   ./backpopulate_xpro_data
   ./backpopulate_mit_edx_data
   ```
- Check that all courses from the following professional offerors have `certification=True`: xpro, csail, ctl, scc, see, mitpe, bootcamps.
- Courses from offeror mitx will have values True or False depending on whether any published runs have an availability value other than "Archived"
